include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -516,7 +516,7 @@ __pattern_merge(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Ran
         __par_backend_hetero::__parallel_merge(_BackendTag{}, ::std::forward<_ExecutionPolicy>(__exec),
                                                ::std::forward<_Range1>(__rng1), ::std::forward<_Range2>(__rng2),
                                                ::std::forward<_Range3>(__rng3), __comp)
-            .wait();
+            .__deferrable_wait();
```

Comment:
I originally thought  `__pattern_inplace_merge` could have an issue, but I believe it is saved by the implicit dependency between kernels from the temporary buffer, so I think nothing needs to be done.

Reply:
ok